### PR TITLE
[Communication][PhoneNumbers] Update SDK to use InternalPhoneNumbersClient

### PR DESCRIPTION
**Pull request by sorrego-msft**
The client uses the InternalPhoneNumbersRestClient directly, this PR will update it so it uses it through the InternalPhoneNumbersClient.

The tests had to be updated so each had a sync and an async version.

### CODE REVIEW

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/SessionRecords/PhoneNumbersClientLiveTests/UpdateCapabilities.json [hunk 1]
```diff
@@ -153,7 +153,7 @@
       }
     },
     {
-      "RequestUri": "https://smstestapp.communication.azure.com/phoneNumbers/\u002BSanitized?api-version=2022-12-01",
+      "RequestUri": "https://smstestapp.communication.azure.com/phoneNumbers/Sanitized/Sanitized",
```
Comment (sorrego-msft) [inline sdk/communication/Azure.Communication.PhoneNumbers/tests/SessionRecords/PhoneNumbersClientLiveTests/UpdateCapabilities.json:156]: I don't know why this change is needed, without it the test throws the error:
```
Message: 
Azure.Core.TestFramework.TestRecordingMismatchException : Unable to find a record for the request GET https://smstestapp.communication.azure.com/phoneNumbers/Sanitized/Sanitized
Uri doesn't match:
    request <https://smstestapp.communication.azure.com/phoneNumbers/Sanitized/Sanitized>
    record  <https://smstestapp.communication.azure.com/phoneNumbers/+Sanitized?api-version=2022-12-01>
```
But it doesn't make sense to me that it sanitize twice

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 2]
```diff
@@ -113,15 +113,12 @@ protected PhoneNumbersClient()
         /// <exception cref="ArgumentNullException"> <paramref name="phoneNumber"/> is null. </exception>
         public virtual async Task<ReleasePhoneNumberOperation> StartReleasePhoneNumberAsync(string phoneNumber, CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNull(phoneNumber, nameof(phoneNumber));
```
Comment (sorrego-msft) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:116]: These Assertions are now done by the InternalPhoneNumbersClient

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 3]
```diff
@@ -288,16 +276,12 @@ public virtual PurchasePhoneNumbersOperation StartPurchasePhoneNumbers(string se
         public virtual async Task<SearchAvailablePhoneNumbersOperation> StartSearchAvailablePhoneNumbersAsync(string twoLetterIsoCountryName, PhoneNumberType phoneNumberType, PhoneNumberAssignmentType phoneNumberAssignmentType,
             PhoneNumberCapabilities capabilities, PhoneNumberSearchOptions options = null, CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNull(twoLetterIsoCountryName, nameof(twoLetterIsoCountryName));
-            Argument.AssertNotNull(capabilities, nameof(capabilities));
```
Comment (sorrego-msft) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:292]: The not null capabilities assert is done by PhoneNumberSearchRequest

**Review by whisper6284 — APPROVED** [review]
